### PR TITLE
Concurrently write time partition

### PR DESCRIPTION
**Pull request by SilverNarcissus**
use concurrently write time partition config to replace max memtable number in each storage group

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapter.java [hunk 1]
```diff
@@ -30,86 +30,70 @@
...
- * The equation: m * Nm + Nm * Ns * Sf * a * c / m + b = S
- * Namely: MemTable data memory size + chunk metadata memory size + static memory size = memory size for write
- *
+ * <p>
+ * Abbreviation of parameters: 1 memtableSize: m 2 maxMemTableNum: Nm 3
+ * maxSeriesNumberAmongStorageGroup: Ns 4 tsFileSizeThreshold: Sf 5 CompressionRatio: c 6 chunk
+ * metadata size: a 7 static memory: b 8 allocate memory for write: S
+ * <p>
+ * The equation: m * Nm + Nm * Ns * Sf * a * c / m + b = S Namely: MemTable data memory size + chunk
+ * metadata memory size + static memory size = memory size for write
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapter.java:61]: format back

Comment (SilverNarcissus) [reply server/src/main/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapter.java]: Sure~

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java [hunk 2]
```diff
@@ -276,7 +275,7 @@ private long getMemtableSizeThresholdBasedOnSeriesNum() {
       return config.getMemtableSizeThreshold();
     }
     long memTableSize = (long) (config.getMemtableSizeThreshold() * config.getMaxMemtableNumber()
-        / IoTDBDescriptor.getInstance().getConfig().getMemtableNumInEachStorageGroup()
+        / IoTDBDescriptor.getInstance().getConfig().getConcurrentWritingTimePartition() / 4
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java:278]: why divide 4?

Comment (jixuan1989) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java:278]: Ok I see...
But you'd better set 4 as a constant value, and refer to the const "4" in  `addOrDeleteStorageGroup()`

File: server/src/test/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapterTest.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ public void addOrDeleteStorageGroup() throws ConfigAdjusterException {
     for (int i = 1; i < 1000000; i++) {
       try {
         IoTDBConfigDynamicAdapter.getInstance().addOrDeleteStorageGroup(1);
-        memTableNum += IoTDBDescriptor.getInstance().getConfig().getMemtableNumInEachStorageGroup();
+        memTableNum += IoTDBDescriptor.getInstance().getConfig().getConcurrentWritingTimePartition() * 4 + 1;
```
Comment (jixuan1989) [inline server/src/test/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapterTest.java:69]: The same issue, set 4 as const and a reference from addOrDeleteStorageGroup()

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java (hunk 2 above)
Comment (SilverNarcissus) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java:278]: Sure~ I will fix it!

File: server/src/test/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapterTest.java (hunk 3 above)
Comment (SilverNarcissus) [reply server/src/test/java/org/apache/iotdb/db/conf/adapter/IoTDBConfigDynamicAdapterTest.java:69]: Sure~ I will fix it!

**Pull request merged**